runtime/near-vm-runner/src/logic/logic.rs

```diff
@@ -2015,6 +2016,152 @@ impl<'a> VMLogic<'a> {
         Ok(())
     }
 
+    /// Creates a promise that will execute a method on the current account with given arguments
+    /// and gas. The created promise will have a special data dependency which can be submitted
+    /// using `promise_yield_resume` within `yield_num_blocks`.
+    ///
+    /// Writes the unique token used to call `promise_yield_resume` to register `register_id`.
+    /// Only the current account is permitted to make the matching call to `promise_yield_resume`.
+    ///
+    /// # Errors
+    ///
+    /// * If `method_name_len + method_name_ptr` or `arguments_len + arguments_ptr` point outside
+    /// the memory of the guest or host returns `MemoryAccessViolation`.
+    /// * If called as view function returns `ProhibitedInView`.
+    ///
+    /// # Returns
+    ///
+    /// Index of the new promise that uniquely identifies it within the current execution of the
+    /// method.
+    ///
+    /// # Cost
+    ///
+    /// The following fees are charged:
+    ///
+    /// * `base` fee;
+    /// * `yield_create_base` fee;
+    /// * `yield_create_byte` for each byte of `method_name` and `arguments`;
+    /// * Fees for reading the `method_name` and `arguments`;
+    /// * Fees for writing the Data ID to the output register;
+    /// * Fees for setting up the receipt.
+    pub fn promise_yield_create(
```

Comment:
@nagisa Should we charge for creation of a data receipt as part of `promise_yield_create`, seeing as one will be created by the protocol if the yield ends up timing out? If so, should we also remove that cost from `promise_yield_resume` given that it was prepaid in the yield create call?

Reply:
Thanks for pointing this out. Indeed, `yield_create_base` fee should cover any additional fees for actions that might occur implicitly after this function has finished executing. In practice what should happen is that the fee reflects the most expensive case of `promise_yield_create` use. I was anticipating of just running the estimator with a contract that never resumes its yields as a point of comparison, but you make a good point that `resume` should not charge at least the base fee for data receipt creation again.